### DISCUSSION TITLE
Converting BigInt to Bytes Issues

### DISCUSSION THREAD

**Discussion by rolandsaven** [graphprotocol/graph-tooling#1386] [Q&A]
So I'm having issues converting a `BigInt` to `Bytes`. What am I doing wrong?

```ts 
let oneString = BigInt.fromString('1');
let one = BigInt.fromI32(1 as i32);

log.info("Bytes.fromBigInt(oneString).toHexString() {}", [Bytes.fromBigInt(oneString).toHexString()]);
log.info("Bytes.fromBigInt(one).toHexString() {}", [Bytes.fromBigInt(one).toHexString()]);    

log.info("oneString.toI32().toString() {}", [oneString.toI32().toString()]);
log.info("one.toI32().toString() {}", [one.toI32().toString()]);
```

The result in MatchStick is 

```
💬 Bytes.fromBigInt(oneString).toHexString() 0x01
💬 Bytes.fromBigInt(one).toHexString() 0x01000000
💬 oneString.toI32().toString() 1
💬 one.toI32().toString() 1
```

So, the `BigInt` that was creates from `i32` has appending zeros when converted to `Bytes`. Why?

**Comment by maoueh** [comment]
Seems to me that there is a problem indeed and I think it's again an endianess problem. I think the problem is maybe `BigInt.fromI32(1 as i32);`,  implementation is:

```
  static fromI32(x: i32): BigInt {
    const byteArray = ByteArray.fromI32(x);
    return BigInt.fromByteArray(byteArray);
  }
```

While `Bytes.fromBigInt` simply case the BigInt into Bytes directly. I would need more checks. Could you share a sample to reproduce your tests?

I would like to try:

```
let oneBytes = ByteArray.fromI32(1 as i32);
let oneReversed = BigInt.fromByteArray(oneBytes.reverse());

log.info("Bytes.fromBigInt(oneReversed).toHexString() {}", [Bytes.fromBigInt(oneReversed).toHexString()]);    
```

> **Note** Untested code

**Reply by rolandsaven** [reply]
@maoueh It's basically an empty subgraph with matchstick and running the script that I pasted above.


I've run your code: (I had to change type though)

```ts
let oneBytes = ByteArray.fromI32(1 as i32);
let oneReversed = BigInt.fromByteArray(changetype<ByteArray>(oneBytes.reverse()));

log.info("Bytes.fromBigInt(oneReversed).toHexString() {}", [Bytes.fromBigInt(oneReversed).toHexString()]); 
```

```
💬 Bytes.fromBigInt(oneReversed).toHexString() 0x00000001
```

**Reply by maoueh** [reply]
Ok so definitely again an endianness problem, not quite sure how we are going to fix that. I say how because I think this will be a breaking change and quite silent.
